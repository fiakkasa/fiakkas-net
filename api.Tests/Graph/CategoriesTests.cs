using HotChocolate.Execution;

namespace api.Tests;

public class CategoriesTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Categories_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  categories {
    totalCount
    nodes {
      associatedCategoryTypes
      createdAt
      href
      id
      internalId
      kind
      title
      updatedAt
      version
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task CategoryById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  categoryById(id: "Q2F0ZWdvcnkKZ2M5ZjU4NzlkNDAxODQ5YTA5YjcxYjQ3OWRkNWRlN2Zm") {
    associatedCategoryTypes
    createdAt
    href
    id
    internalId
    kind
    title
    updatedAt
    version
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
