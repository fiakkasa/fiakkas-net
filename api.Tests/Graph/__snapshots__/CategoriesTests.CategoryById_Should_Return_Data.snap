{
  "data": {
    "categoryById": {
      "associatedCategoryTypes": [],
      "createdAt": "2024-01-01T00:00:00.000Z",
      "href": null,
      "id": "Q2F0ZWdvcnkKZ2M5ZjU4NzlkNDAxODQ5YTA5YjcxYjQ3OWRkNWRlN2Zm",
      "internalId": "c9f5879d-4018-49a0-9b71-b479dd5de7ff",
      "kind": "NONE",
      "title": "Title",
      "updatedAt": null,
      "version": 1
    }
  }
}
